### PR TITLE
Fixes #16228: Fetching nodes last run is not batched

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16228

### CODE REVIEW

**Comment by ncharles** [comment]
Not tested yet, do not merge

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionRepositoryImpl.scala [hunk 1]
```diff
@@ -84,63 +86,68 @@ case class RoReportsExecutionRepositoryImpl (
...
+              case tuple@(r, t1, t2, t3, t4, t5) => (r, unserNodeConfig(t1, t2, t3, t4, t5))
+            }.to[Vector]
+          } yield {
+
+            val runsMap = (runs.map { case (r, optConfig) =>
+              val run = r.toAgentRun
+              val config = run.nodeConfigVersion.map(c => (c, optConfig))
+              (run.agentRunId.nodeId, AgentRunWithNodeConfig(run.agentRunId, config, run.isCompleted, run.insertionId))
+            }).toMap
+            val truc = ids.map(id => (id, runsMap.get(id))).toMap
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionRepositoryImpl.scala]: truc remaining

**Comment by ncharles** [comment]
Commit modified

**Comment by fanf** [comment]
OK, merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**